packages/ui-extensions/src/surfaces/admin/components/Badge/Badge.ts

```diff
@@ -0,0 +1,22 @@
+import {createRemoteComponent} from '@remote-ui/core';
+import type {AccessibilityLabelProps, SizeScale, Tone} from '../shared';
+import {IconName} from '../Icon/IconName';
+
+interface BadgeBaseProps extends AccessibilityLabelProps {
+  tone?: Tone;
+  size?: Extract<SizeScale, 'small-100' | 'base'>;
+}
+
+interface BadgeIconProps {
+  icon: IconName;
+  iconPosition?: 'start' | 'end';
+}
+
+interface BadgeNoIconProps {
+  icon: never;
+  iconPosition: never;
+}
+
+export type BadgeProps = BadgeBaseProps & (BadgeIconProps | BadgeNoIconProps);
```

Comment:
This doesn't look great in the generated docs. Looking into a better way to achieve the same result without the awkward union showing up in the docs.

Reply:
I don't think there's another way to represent this in typescript unfortunately. We have the same problem with `Image` docs